extensions.html

```diff
@@ -206,28 +233,82 @@ <h2><dfn>GamepadHapticActuator</dfn> Interface</h2>
-        The actuator type determines the force applied for a given
-        <var>value</var> in <code>GamepadHapticActuator.pulse()</code>.
+        The actuator type describes the capabilities of the gamepad's
+        haptics configuration.
       </p>
 
       <pre class="idl">
         enum GamepadHapticActuatorType {
-          "vibration"
+          "dual-rumble"
         };
       </pre>
 
       <dl data-dfn-for="GamepadHapticActuatorType">
-        <dt><dfn>vibration</dfn></dt>
+        <dt><dfn>dual-rumble</dfn></dt>
         <dd>
-          Vibration is a rumbling effect often implemented as an offset weight
-          driven on a rotational axis. The <code>value</code> of a vibration
-          force determines the frequency of the rumble effect and is
-          normalized between <code>0.0</code> and <code>1.0</code>. The
-          neutral value is <code>0.0</code>.
+          Dual-rumble describes a haptics configuration with an eccentric
+          rotating mass (ERM) vibration motor in each handle of a standard
+          gamepad. In this configuration, either motor is capable of vibrating
+          the whole gamepad. The ERM masses are unequal so that the effects of
+          each can be combined to create more complex haptic effects. A device
+          may expose a vibration actuator with type <code>'dual-rumble'</code>
+          if it has a dual-ERM configuration, or if it has an alternate haptics
+          configuration that is compatible with dual-rumble effects.
         </dd>
       </dl>
     </section>
 
+    <section>
+      <h2><dfn>GamepadHapticEffectType</dfn> Enum</h2>
+      <p>
+        The effect type defines how the effect parameters are interpreted by
+        the actuator.
+      </p>
+
+      <pre class="idl">
+        enum GamepadHapticEffectType {
+          "dual-rumble"
+        };
+      </pre>
+
+      <dl data-dfn-for="GamepadHapticEffectType">
+        <dt><dfn>dual-rumble</dfn></dt>
+        <dd>
+          A dual-rumble effect is a fixed-length, constant-intensity vibration
+          effect intended for an actuator of type <code>'dual-rumble'</code>.
+          Dual-rumble effects are defined by four parameters.
+          <var>duration</var> sets the duration of the vibration effect in
+          milliseconds, default 0. <var>startDelay</var> sets the duration of
+          the delay after <code>playEffect()</code> is called until vibration
+          is started, default 0. During the delay interval, the actuator must
+          not vibrate. To prevent abuse, the user agent should consider an
+          effect invalid if the total effect duration (the sum of the vibration
+          interval and delay interval) exceeds some reasonable value, say 5
+          seconds. <var>strongMagnitude</var> and <var>weakMagnitude</var> set
+          the vibration intensity levels for the heavier and lighter ERM
+          motors, normalized to the range <code>[0,1]</code>, default 0.
+        </dd>
```

Comment:
Is it valid for any of these parameters to be 0?

Reply:
Yes, 0 is valid for all parameters. A dual-rumble effect with the default parameters describes a 0-length effect with no vibration from either motor. In practice it behaves the same as calling reset().